pyupgrade.py

```diff
@@ -1880,6 +1889,70 @@ def _replace(i, mapping, node):
         elif token.offset in visitor.os_error_alias_simple:
             node = visitor.os_error_alias_simple[token.offset]
             _replace(i, collections.defaultdict(lambda: 'OSError'), node)
+        elif token.offset in visitor.os_error_alias_excepts:
+            line, utf8_byte_offset = token.line, token.utf8_byte_offset
+
+            # find all the arg strs in the tuple
+            except_index = i
+            while tokens[except_index].src != 'except':
+                except_index -= 1
+            start = _find_open_paren(tokens, except_index)
+            func_args, end = _parse_call_args(tokens, start)
+
+            # save the exceptions and remove the block
+            arg_strs = [_arg_str(tokens, *arg) for arg in func_args]
+            del tokens[start:end]
+
+            # rewrite the block without dupes
+            args = []
+            seen_os_error = False
+            for arg_str in arg_strs:
+                left, part, right = arg_str.partition('.')
+                if (
+                        left in visitor.OS_ERROR_ALIAS_MODULES and
+                        part == '.' and
+                        right == 'error'
+                ):
+                    if not seen_os_error:
+                        args.append('OSError')
+                    seen_os_error = True
+                elif (
+                        left in visitor.OS_ERROR_ALIASES and
+                        part == right == ''
+                ):
+                    if not seen_os_error:
```

Comment:
could this be simplified with a `set` and `list`?

Reply:
I wanted to keep the order, will use `list(OrderedDict.from_keys(...))`